src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilder.cs

```diff
@@ -2,72 +2,72 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.ComponentModel;
 using Microsoft.AspNetCore.Connections;
 using Microsoft.AspNetCore.SignalR.Internal.Protocol;
-using Newtonsoft.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public class HubConnectionBuilder : IHubConnectionBuilder
     {
-        private readonly Dictionary<KeyValuePair<string, Type>, object> _settings = new Dictionary<KeyValuePair<string, Type>, object>();
-        private Func<IConnection> _connectionFactoryDelegate;
+        private ServiceProvider _serviceProvider;
 
-        public void ConfigureConnectionFactory(Func<IConnection> connectionFactoryDelegate) =>
-            _connectionFactoryDelegate = connectionFactoryDelegate;
+        public IServiceCollection Services { get; }
 
-        public void AddSetting<T>(string name, T value)
+        public HubConnectionBuilder()
         {
-            _settings[new KeyValuePair<string, Type>(name, typeof(T))] = value;
-        }
-
-        public bool TryGetSetting<T>(string name, out T value)
-        {
-            value = default;
-            if (!_settings.TryGetValue(new KeyValuePair<string, Type>(name, typeof(T)), out var setting))
+            Services = new ServiceCollection();
+            Services.AddSingleton<HubConnection>(serviceProvider =>
```

Comment:
Right now this is a singleton. Should calling `Build()` on a HubConnectionBuilder multiple times return different instances?